src/js/common/schemaform/Address.jsx

```diff
@@ -42,14 +42,24 @@ class Address extends React.Component {
   }
 
   handleChange(path, update) {
-    let newState = set(path, update, this.props.formData);
+    const formData = this.props.formData
+      ? this.props.formData
+      : getDefaultFormState(this.props.schema, undefined, this.props.registry.definitions);
+
+    let newState = set(path, update, formData);
 
     // if country is changing we should clear the state
     if (path === 'country') {
       newState = set('state', undefined, newState);
     }
 
-    this.props.onChange(newState);
+    const fields = Object.keys(newState);
+    if (fields.length === 0 || fields.every(field => field === 'country' || newState[field] === undefined)) {
+      // send undefined so that the object is removed
```

Comment:
This is related to your above comment, too. I'm doing this to make sure we return `undefined` whenever we're in a state where the user has entered no form data. This keeps us from sending empty objects to the backend unnecessarily which can cause validation problems.

This particular code is catching the case where a user makes a change to an address field, then removes it. If the form data has only `undefined` fields, except for a country that matches the default, then return `undefined` instead of the object, because we know there's no address info in it.